gcc/testsuite/gcc.target/arc64/cmodel-large-cpp-thunk.cpp

```diff
@@ -0,0 +1,26 @@
+/* Check that post-reload thunk generation does not try to create new pseudos. */
+/* { dg-do compile } */
+/* { dg-options "-O -mcmodel=large" } */
+
+#include <cstddef>
+
+#define NO_INLINE __attribute__((noinline))
+
+class A {
+  virtual void foo() = 0;
+};
+class B {
+  virtual void bar() = 0;
+};
+class C : public A, public B {
+  NO_INLINE void foo() {}
+  NO_INLINE void bar() {}
+};
+
+C c;
+
+int main() {
```

Comment:
Could we make this a run test in addition to compile? You'd need to make foo() and bar() output something, call those functions from an instance of class C, then use the "dg-do run" and "dg-output" directives to check for correctness. For an example of the latter, see gcc/testsuite/c-c++-common/ubsan/div-by-zero-1.c.